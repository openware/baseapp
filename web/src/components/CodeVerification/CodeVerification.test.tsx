import { shallow } from 'enzyme';
import * as React from 'react';
import { CodeVerification, CodeVerificationProps } from '../';
import { IntlProvider } from 'react-intl';

const defaultProps: CodeVerificationProps = {
    placeholder: '',
    type: '',
    codeLength: 6,
    code: '',
    onChange: jest.fn(),
    onSubmit: jest.fn(),
};

const setup = (props: Partial<CodeVerificationProps> = {}) => shallow(<IntlProvider locale="en"><CodeVerification {...{ ...defaultProps, ...props }} /></IntlProvider >);

describe('CodeVerification test', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toBeDefined();
    });
});
