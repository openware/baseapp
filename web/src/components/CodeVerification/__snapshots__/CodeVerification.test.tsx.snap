// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeVerification test should render 1`] = `
<ContextProvider
  value={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "defaultRichTextElements": undefined,
      "formatDate": [Function],
      "formatDateTimeRange": [Function],
      "formatDateToParts": [Function],
      "formatDisplayName": [Function],
      "formatList": [Function],
      "formatListToParts": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getDisplayNames": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": undefined,
      "wrapRichTextChunksInFragment": undefined,
    }
  }
>
  <CodeVerification
    code=""
    codeLength={6}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
    placeholder=""
    type=""
  />
</ContextProvider>
`;
